tests/testthat/helper.R

```diff
@@ -1,16 +1,37 @@
 Sys.setlocale("LC_COLLATE", "C") ## What CRAN does
 set.seed(666)
 
-## Our "test package" common harness code
-## `try` added because of some devtools::document weirdness
-try({
-    crunch_test_path <- system.file("crunch-test.R", package="crunch")
-    if (crunch_test_path == "") {
-        # hack for loadall
-        crunch_test_path <- "../../inst/crunch-test.R"
+# find a file starting with the current directory and looking (recursively) in parent folders limited to levels_up
+find_file <- function (file_name, levels_up = 3) {
+    for (i in 0:levels_up) {
+        if (i == 0) {
+            pth <- "."
+            
+        } else {
+            pth <- file.path(".", paste0(rep("..", i), collapse = "/"))
+        }
+        fls <- list.files(pth, recursive = TRUE)
+        fl <- fls[grepl(file_name, fls)]
+        if (length(fl) == 1) return(file.path(pth,fl))
+        if (length(fl)  > 1) {
+            warning("More than one file named ", file_name, " using the first one.")
+            return(file.path(pth,fl[1]))
+        }
     }
+    stop("Found no file named ", file_name, " Stopping tests because they will not function without ", file_name)
+}
+
+## Our "test package" common harness code
+crunch_test_path <- system.file("crunch-test.R", package="crunch")
+if (crunch_test_path == "") {
+    # hack for loadall
+    crunch_test_path <- find_file("crunch-test.R")
+}
+# only source crunch-test.R when the session is not interactive
```

Comment:
Why?

Reply:
The main reason for this is that when `crunch-test.R` is sourced, it alters the current global environment. The way that `load_all` works, for better or worse, will always execute all of the helper files (my understanding is the reason for this is so that those helper functions are available to help you write and debug tests). But the main consequence is that when Gordon and I are working on something and we want to be connected to a host other than local, `source`ing `crunch-test.R` (via `devtools::load_all()`) clobbers those settings. (And adds a bunch of stuff that we don't need for debugging to our global environments). 

I know it sounds minor, but I go back and forth and comment this source line many times a day as I'm developing.

There are two other approaches we might have to take:
1. Instead of just testing interactivity, we can test what the ultimate caller is, and not source this file when what is calling it is not `devtools::load_all()`. I've tested this with 
```quote(devtools::load_all) != sys.calls()[[1]][[1]]```
and it works well (even calling `devtools::test()` interactively still loads, running tests via make, via RStudio's test, etc. all work (and necessarily load the helper file). I would want to make this a bit more robust in case for example: some how `sys.calls` doesn't have those indices we still get a `TRUE` like value instead of `logical(0)`.

1. We could make `crunch-test.R` be more respectful of user/developer environments and options and move the code that alters the global environment to a function that is called before the tests are loaded but critically not somewhere that `devtools::load_all()` will run it (ie in `tests/testthat.r` for `R CMD check`, in the `Makefile` strings that run tests for `make test`, in the `crunchdev strings that run tests for anything that uses the `crunchdev` helpers, finally, I don't know of a way to insert into what Rstudio is doing when it runs `devtools::test(".")` which is a pretty big drawback.) 

Conceptually I prefer the second one since it is more respectful of user/developer options and global environments, but given the added hassle it would be to run the setup code/function when it would need to be run plus that it would be a pretty major change in the way that code works, I'm opting for the first (slash 0th — using `interactive()` — option).